components/encoder/single/single_encoder.go

```diff
@@ -182,10 +184,13 @@ func (e *Encoder) Reconfigure(
 // Start starts the Encoder background thread.
 func (e *Encoder) Start(ctx context.Context) {
 	encoderChannel := make(chan board.Tick)
-	e.I.AddCallback(encoderChannel)
+	err := e.board.StreamTicks(e.cancelCtx, []string{e.diPinName}, encoderChannel, nil)
```

Comment:
I think you might need to close the encoder channel when the loop in Start returns. Or is the cancel function and active background worker sufficient in `Close`? Is the server shutting down with no goroutines leaking?

What about `go test -race` on the driver packages?